src/sync/sync.nim

```diff
@@ -0,0 +1,238 @@
+  for exercise in exercises:
+    let slug = exercise.slug.string
+    let trackDocsDir = joinPath(trackPracticeExercisesDir, slug, ".docs")
+
+    if dirExists(trackDocsDir):
+      let psExerciseDir = psExercisesDir / slug
+      if dirExists(psExerciseDir):
+
+        # If the exercise in problem-specifications has an `introduction.md`
+        # file, the track exercise must have a `.docs/introduction.md` file.
+        let introFilename = "introduction.md"
+        let psIntroPath = psExerciseDir / introFilename
+        if fileExists(psIntroPath):
+          let trackIntroPath = trackDocsDir / introFilename
+          if fileExists(trackIntroPath):
+            checkFilesIdentical(psIntroPath, trackIntroPath, slug,
+                               introFilename, seenUnsynced, conf, result)
+          else:
+            logNormal(&"[error] {slug}: {introFilename} is missing")
+            seenUnsynced.incl skDocs
+
+        # The track exercise must have a `.docs/instructions.md` file.
+        # Its contents should match those of the corresponding `instructions.md`
+        # file in problem-specifications (or `description.md` if that file
+        # doesn't exist).
+        let instrFilename = "instructions.md"
+        let trackInstrPath = trackDocsDir / instrFilename
+        if fileExists(trackInstrPath):
+          let descFilename = "description.md"
+          let psInstrPath = psExerciseDir / instrFilename
+          let psDescPath = psExerciseDir / descFilename
+          if fileExists(psInstrPath):
+            checkFilesIdentical(psInstrPath, trackInstrPath, slug,
+                               instrFilename, seenUnsynced, conf, result)
+          elif fileExists(psDescPath):
+            checkFilesIdentical(psDescPath, trackInstrPath, slug,
+                                instrFilename, seenUnsynced, conf, result)
+          else:
+            logNormal(&"[error] {slug}: does not have an upstream " &
+                      &"{instrFilename} or {descFilename} file")
+            seenUnsynced.incl skDocs
+        else:
+          logNormal(&"[error] {slug}: {instrFilename} is missing")
+          seenUnsynced.incl skDocs
+
+      else:
+        logDetailed(&"[skip] {slug}: does not exist in problem-specifications")
+    else:
+      logNormal(&"[error] {slug}: .docs dir missing")
+      seenUnsynced.incl skDocs
+
+proc checkFilepaths(exercises: seq[Exercise], seenUnsynced: var set[SyncKind]) =
+  if false:
+    seenUnsynced.incl skFilepaths
+
+type
+  MetadataPair = object
+    trackJsonPath: string
+    updatedJson: JsonNode
+
+proc checkMetadata(exercises: seq[Exercise],
+                   psExercisesDir: string,
+                   trackPracticeExercisesDir: string,
+                   seenUnsynced: var set[SyncKind],
+                   conf: Conf): seq[MetadataPair] =
+  for exercise in exercises:
+    let slug = exercise.slug.string
+    let trackMetaDir = joinPath(trackPracticeExercisesDir, slug, ".meta")
```

Comment:
```suggestion
    let exerciseMetaDir = joinPath(trackPracticeExercisesDir, slug, ".meta")
```

Reply:
This particular name isn't there anymore. I've used `metaDir` sometimes.